### DISCUSSION TITLE
Mesh refinement

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#25774] [Q&A Meshing] [answered]
All,

I know "uniform_refine" refines a mesh in both horizonal and vertical directions in the 2D mesh but I would like to refine a mesh in the only vertical direction. 
Please let me know if this is possible on MOOSE.
S. Thomas

**Comment by GiudGiud** [comment]
Hello

Not possible afaik.
You ll have to bias your mesh in one direction when you generate it if you want to get this non-uniformity

Guillaume

**Reply by styyokuda** [reply, marked as answer]
Thank you, Guillaume

**Reply by lindsayad** [reply]
Anisotropic refinement has been a wish list item for @roystgnr for a long time in libMesh. Just hasn’t happened yet!

**Reply by roystgnr** [reply]
Worse: this isn't one of the "hasn't happened yet because I haven't gotten around to it" wishlist items, this is a "hasn't happened yet because the very concept would break other library code's assumptions in a hundred places" wishlist item.  (Not hyperbole - I counted 100+ trouble spots when first looking into this long ago, and I'm sure it's gone up since.)

If someone held a gun to my head and demanded anisotropic adaptive refinement next month, I'd start working on metric-tensor-based triangle/tet mesh regeneration, I wouldn't even bother trying to anisotropic get quad/hex division working.  I've run simulations on loads of very anisotropic meshes for flows with thin boundary layers, but the way to get them is via the initial mesh generation, I'm afraid.
